zapcore/console_encoder.go

```diff
@@ -103,52 +102,52 @@ func (c consoleEncoder) EncodeEntry(ent Entry, fields []Field) ([]byte, error) {
 	// If there's no stacktrace key, honor that; this allows users to force
 	// single-line output.
 	if ent.Stack != "" && c.StacktraceKey != "" {
-		line.WriteByte('\n')
-		line.WriteString(ent.Stack)
+		line.AppendByte('\n')
```

Comment:
That might not be so bad since then we'll implement `WriteString` with a return of `(int, error) which will allow `io.WriteString` to efficiently write a string to the buffer without converting to `[]byte`.